src/internal/components/pluggableVisualizations/geoChart/PluggableGeoPushpinChart.tsx

```diff
@@ -0,0 +1,103 @@
+import { setGeoPushPinUiConfig } from "../../../utils/uiConfigHelpers/geoPushpinChartUiConfigHelper";
+import { removeSort } from "../../../utils/sort";
+import { getReferencePointWithSupportedProperties } from "../../../utils/propertiesHelper";
+import { VisualizationTypes } from "../../../../constants/visualizationTypes";
+
+import UnsupportedConfigurationPanel from "../../configurationPanels/UnsupportedConfigurationPanel";
+
+export class PluggableGeoPushpinChart extends PluggableBaseChart {
+    constructor(props: IVisConstruct) {
+        super(props);
+        this.type = VisualizationTypes.PUSHPIN;
+        this.supportedPropertiesList = [];
+        this.initializeProperties(props.visualizationProperties);
+    }
+
+    public getExtendedReferencePoint(referencePoint: IReferencePoint): Promise<IExtendedReferencePoint> {
+        const clonedReferencePoint = cloneDeep(referencePoint);
+        let newReferencePoint: IExtendedReferencePoint = {
+            ...clonedReferencePoint,
+            uiConfig: cloneDeep(GEO_PUSHPIN_CHART_UICONFIG),
+        };
+
+        const buckets = get(clonedReferencePoint, BUCKETS, []);
+        const locations = getBucketItemsByType(buckets, bucketNames.LOCATION, [GEO_ATTRIBUTE]);
+        const measuresSize = getBucketItemsByType(buckets, bucketNames.SIZE, [METRIC]);
+        const measuresColor = getBucketItemsByType(buckets, bucketNames.COLOR, [METRIC]);
+        const segments = getPreferredBucketItems(
+            buckets,
+            [bucketNames.STACK, bucketNames.SEGMENT_BY],
+            [GEO_ATTRIBUTE, ATTRIBUTE],
+        );
+
+        set(newReferencePoint, BUCKETS, [
+            {
+                localIdentifier: bucketNames.LOCATION,
+                items: locations,
+            },
+            {
+                localIdentifier: bucketNames.SIZE,
+                items: measuresSize,
+            },
+            {
+                localIdentifier: bucketNames.COLOR,
+                items: measuresColor,
+            },
+            {
+                localIdentifier: bucketNames.SEGMENT_BY,
+                items: segments,
+            },
+        ]);
+
+        newReferencePoint = setGeoPushPinUiConfig(newReferencePoint, this.intl, this.type);
+        newReferencePoint = configurePercent(newReferencePoint, false);
+        newReferencePoint = configureOverTimeComparison(newReferencePoint);
+        newReferencePoint = getReferencePointWithSupportedProperties(
+            newReferencePoint,
+            this.supportedPropertiesList,
+        );
+        newReferencePoint = removeSort(newReferencePoint);
+
+        return Promise.resolve(sanitizeFilters(newReferencePoint));
+    }
+
+    protected renderConfigurationPanel() {
+        const configPanelElement = document.querySelector(this.configPanelElement);
+        if (configPanelElement) {
+            const properties: IVisualizationProperties = get(
+                this.visualizationProperties,
+                "properties",
```

Comment:
it could be
```
const properties: IVisualizationProperties = get(
                this,
                "visualizationProperties.properties",
```